### DISCUSSION TITLE
Add prefix to img src attribute in editor (but not in HTML)

### DISCUSSION THREAD

**Discussion by webdesq** [ueberdosis/tiptap#1244] [General Questions] [answered]
Hi, we're evaluating TipTap in a svelte project. Everything works great!
However, we have a fairly specific requirement. All URLs to images (the IMG src attribute) should be prefixed when using the editor. (like ```<IMG src="/myprefix/images/duck.jpg">```) However, the resulting HTML should not contain this prefix (```<IMG src="/images/duck.jpg">```).

Any clues on how we could implement this? Is there some event we can hook into when IMAGE nodes are rendered in the editor?

**Comment by hanspagel** [comment, marked as answer]
You could extend the Image extension with a node view:
https://www.tiptap.dev/guide/node-views/js

You have full control over the rendering inside the editor (while editing) then, for example to add a prefix. But the HTML output will still be the same (without the prefix).

Hope that helps!

**Reply by webdesq** [reply]
Thanks Hans!

It works nicely now.
